src/wallet/wallet.cpp

```diff
@@ -626,6 +626,15 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
 
         Lock();
         Unlock(strWalletPassphrase);
+
+        // if we are using HD, replace the HD master key with a new one
+        if (!hdChain.masterKeyID.IsNull()) {
+            CKey key;
+            key.MakeNewKey(true);
+            if (!SetHDMasterKey(key))
```

Comment:
SetHDMasterKey() can only return true. When it fails it throws an exception. Do we need to catch that exception?